silo-vaults/contracts/interfaces/IIncentivesDistributionSolution.sol

```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity >=0.5.0;
+
+/// @title Incentives Distribution Solution interface
+interface IIncentivesDistributionSolution {
+    /// @notice Called after a token transfer.
+    /// @dev Notifies the solution about the token transfer.
+    function afterTokenTransfer(
+        address _sender,
+        uint256 _senderBalance,
+        address _recipient,
+        uint256 _recipientBalance,
+        uint256 _totalSupply,
+        uint256 _amount
+    ) external;
+}
```

Comment:
let's add `incentivesDistributionSolutionPing() returns bytes32` method.

Reply:
I started implementation, and I see that 'incentivesDistributionSolutionPing' needs to be added to ve-silo gauges (no harm, but for some reason, I don't feel that we need it). When we are adding claiming logics or incentives distribution we can call "checkpoint" method on a vault (I think we will have one and it will be permission less). If there is any problem with a logic or a distribution solution it will revert. Wdyt?

Reply:
if Aiham will be adding some logic, then how would you decide, this logic is for gauge and you need to call `checkpoint`?

check `(address != 0)` is not enough in my opinion, if we know this will be manage by human, it is easy to set some invalid address.

Reply:
We decided not to do pings for v2. Regarding `checkpoint` I meant Vault.